src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java

```diff
@@ -121,7 +121,15 @@ public Response createLibraryCardDaaRelation(
           .filter(card -> card.getInstitutionId() == authedUser.getInstitutionId().intValue())
           .findFirst();
       if (matchingCard.isEmpty()) {
-        return Response.status(Status.NOT_FOUND).build();
+        LibraryCard lc = new LibraryCard();
+        lc.setUserId(user.getUserId());
+        lc.setInstitutionId(authedUser.getInstitutionId());
+        lc.setEraCommonsId(user.getEraCommonsId());
+        lc.setUserName(user.getDisplayName());
+        lc.setUserEmail(user.getEmail());
+        lc.setCreateUserId(authedUser.getUserId());
+        LibraryCard createdLc = libraryCardService.createLibraryCard(lc, user);
+        matchingCard = Optional.of(createdLc);
```

Comment:
I think this functionality would make more sense in the service.

Reply:
Do you mean the DAAService or the LibraryCardService?

Reply:
Probably the LibraryCardService, since it doesn't involve any DAA functionality directly.